### ISSUE TITLE
Track subspace usage

### ISSUE THREAD

**Issue by novocaine** [vector-im/element-meta#208]
We currently track the number of spaces a user is in via a [user property](https://github.com/matrix-org/matrix-analytics-events/blame/0101e4fd25ded5fb2cba8a9119cb061e36296369/schemas/UserProperties.json#L15).

We'd like to extend this to track the number of subspaces a user is in. We're interested in this to determine whether our users are finding subspaces useful and how well to support them in future.

For a space A to be considered a subspace either of the following can be true, from [here](https://github.com/matrix-org/matrix-spec-proposals/blob/old_master/proposals/1772-groups-as-rooms.md#relationship-between-rooms-and-spaces)

* The space room A contains `m.space.parent` pointing to some parent space room B
* Another space room B contains `m.space.child` pointing to this space room A. Note that the user won't have visibility of this unless they have joined space room B.

Add as user properties:

`numSubSpaces` - the number of spaces that are subspaces the user is currently in. 

`numRoomsSpaceDepth0` - the number of rooms the user is in that are the immediate children of a space the user is in, and the space is not a subspace (i.e. has no known parent spaces)
`numRoomsSpaceDepth1` - as above, but where the space is a subspace (has at least one known parent space) with depth 1
`numRoomsSpaceDepth2` - as above, but with depth 2
`numRoomsSpaceDepth3orGreater` - as above, with depth 3 or greater

**Comment by gileluard** [comment]
We can do it with the actual implementation of the iOS SDK but I'm not sure we can reliably track the depth of a room if the user didn't join all the subspaces of the subspace path for a room. Need to be confirmed with Mickael I think.

**Comment by niquewoodhouse** [comment]
I think we'd want to if possible know if people are using sub spaces via the UI too.

So, in the sidebar, are people expanding a space and selecting sub spaces? I worry that just tracking if people are in subspaces, we're not finding out if they are using them on mobile as a feature.  They might have joined via web for example.

**Comment by gileluard** [comment]
Agreed. I cannot help for this part as UI support for subspaces have not been implemented on iOS.
